crates/noirc_frontend/src/node_interner.rs

```diff
@@ -52,6 +57,12 @@ pub struct NodeInterner {
     // methods from impls to the type.
     structs: HashMap<StructId, Shared<StructType>>,
 
+    // Type Aliases map.
+    //
+    // Map type aliases to the actual type.
+    // When resolving types, check against this map to see if a type alias is defined.
+    type_aliases: Vec<Rc<RefCell<TypeAliasType>>>,
```

Comment:
```suggestion
    type_aliases: Vec<TypeAliasType>,
```
Why do we need these to be mutably shared? Can we not just retrieve a mutable reference to them when needed?

Reply:
You may have to move around some things so that the reference lifetimes no longer intersect. I'd recommend changing that segment to a free function first then experiment with moving code around.